### PR TITLE
[FEAT] second teacher

### PR DESCRIPTION
**Pull request by hasan-sh**
**Description**
#3610 

Depends on #4681

**Fixes _issue or discussion number_**

**How to test**
go to classes and invtie second teachers.

**Checklist**
Done? Check if you have it all in place using this list: (mark with x if done)

- [x] invite/remove second teacher
- [x] update `profile` page
- [x] update `class-overview` page
- [x] update `customize-class` page
- [x] update `customize-adventure` page
- [x] show programs of each teacher in class
- [x] show adventures of each teacher in class
- [ ] add tests
    - [x] could invite teachers
    - [x] could customize and edit class
    - [x] could only view adventures
    - [ ] could only view programs

**Summary: Updates and Rules**

- second teachers can be added from the `/fore-teachers/class/[class_id]` page.
- only teachers can be second teachers
- second teachers can do everything the main teacher can do except deleting the class.
- second teachers can create, read and update a class
- upon creating adventures the ownership is simply to the creator (current user)
- adventures of all second teachers in all classes of the current teacher are visible to the teacher
- adventures of teachers are visible to second teachers
- second teacher cannot see other second teachers' adventures unless they are in the same class.
- adventures can only be updated/deleted by the creator.
- second teachers can customize any class they are in (granting them the same rights as the respective primary teacher.)

If you're unsure about any of these, don't hesitate to ask. We're here to help!

### CODE REVIEW

File: content/adventures/en.yaml [hunk 1]
```diff
@@ -859,7 +859,7 @@ adventures:
                     ```
```
Comment (Felienne) [inline content/adventures/en.yaml:859]: Hmmm these changes are going to upset weblate probably, but also... why are they in this PR? These seem uprelated? I don't know how they got here, do you understand @hasan-sh?

Comment (Felienne) [reply content/adventures/en.yaml:859]: (maybe a merge from main when the conflicts are addressed will solve this?)

File: README.md [hunk 2]
```diff
@@ -5,7 +5,7 @@
 ![](https://img.shields.io/github/issues/hedyorg/hedy)
 
 ![](static/images/Hedy-logo-96x96-round.png)
-
+<div id="top"></div>
```
Comment (Felienne) [inline README.md:8]: This too is a recent change on main, and as fas as I understand should not be here.

**Comment by Felienne** [comment]
I have correctly merged main in @hasan-sh! I set it to draft because it is now missing your new strings and translations. 

Can you add those by running extract again? (if not no problem, than I will finish it later)

**Comment by hasan-sh** [comment]
> Can you re-add them?

Sure, will do so tomorrow morning ;)

**Review by Felienne — CHANGES_REQUESTED** [review]
Hi @hasan-sh!

I tried the feature and it works great!! Teachers will be very happy!

I love also how you have thought of many details, like adding the role in the invite list:

image

There are still a few tests failing, hope you can look into those! 

In addition to that, a few comments from my first trial run.

* I am not sure if I am supposed to type a username or email address here? Or maybe both work? Can we add a label?
image

* inviting a non-existing teacher gives a weird error:
image

* this is a bit of a "stretch goal" (let's do that in a separate PR!) but maybe we want to show the teachers here too: 
image
 
* this too is a nice to have, but maybe after the adventure name here we can also add the author (maybe between brackets?) 
image

* I was not sure what this label is:
image
 (but maybe I'd understand if the strings are filled)

**Comment by Felienne** [comment]
Hi @hasan-sh! I think in principle, this is good to go (it seems to work well) but we need to look at the tests, do they catch a real problem, are they breaking for no reason (that sometimes happens) or are they in need of some changes due to the PR?

Let us know if you need help fixing the tests!

**Comment by hasan-sh** [comment]
> Hi @hasan-sh! I think in principle, this is good to go (it seems to work well) but we need to look at the tests, do they catch a real problem, are they breaking for no reason (that sometimes happens) or are they in need of some changes due to the PR?
> 
> Let us know if you need help fixing the tests!

I changed my approach on testing the feature; some tests relied on others. Now they run independenlty! All tests passed locally! However, the error that the github action yields is: `Os is not defined`. I think it's some missing env variable or some other small thing!

**Comment by Felienne** [comment]
> > Hi @hasan-sh! I think in principle, this is good to go (it seems to work well) but we need to look at the tests, do they catch a real problem, are they breaking for no reason (that sometimes happens) or are they in need of some changes due to the PR?
> > Let us know if you need help fixing the tests!
> 
> I changed my approach on testing the feature; some tests relied on others. Now they run independenlty! All tests passed locally! However, the error that the github action yields is: `Os is not defined`. I think it's some missing env variable or some other small thing!

For me also the e2e tests pass locally (I dud not try Cypress) so indeed I guess something small/weird is amiss, but we still need to see how to get this fixed since we don't want to merge broken tests. I don't know what the issue is myself.

Do have time to dive in and do you see a path forward? Because of all the changes and commits it is very hard for someone else to understand what happens (or maybe @jpelay has a brilliant plan?)

**Comment by hasan-sh** [comment]
> Do have time to dive in and do you see a path forward? Because of all the changes and commits it is very hard for someone else to understand what happens (or maybe @jpelay has a brilliant plan?)

Yes, I do. Just pushed the tests for adding tags to adventures! Will look into this!

**Comment by Felienne** [comment]
Woohoo, all good to go! I will do Weblate first and then this one.

**Review by Felienne — APPROVED** [review]

**Pull request merged**